source/tools/hash_utils.h

```diff
@@ -24,9 +26,22 @@ namespace emp {
   //https://www.boost.org/doc/libs/1_37_0/doc/html/hash/reference.html#boost.hash_combine
   constexpr inline std::size_t hash_combine(std::size_t hash1, std::size_t hash2)
   {
-    return hash1 ^ (hash2 * 0x9e3779b9 + (hash1 << 6) + (hash1 >> 2));
+    return hash1 ^ (hash2 + 0x9e3779b9 + (hash1 << 6) + (hash1 >> 2));
   }
 
+  template <typename Container, size_t Seed = 0>
+  struct ContainerHash
+  {
+    size_t operator()(const Container& v) const {
+        size_t data = Seed;
+        for (const auto& x : v) {
+            using T = typename std::decay<decltype(x)>::type;
```

Comment:
more descriptive than T?

Reply:
in 7796005, changed to `item_type` to correspond with `item` from previous review